### PR TITLE
fix #5028: return null if AssertNumRows input empty

### LINKED ISSUE #5028: query result error
### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS w5;
CREATE DATABASE w5;
USE w5;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NOT NULL,c_0_1 DATE NOT NULL,c_0_2 VARCHAR(11) NOT NULL,c_0_3 TINYINT NOT NULL,c_0_4 VARCHAR(21) NOT NULL,c_0_5 DOUBLE NOT NULL,c_0_6 DATETIME NOT NULL,c_0_7 ARRAY<STRING> NOT NULL,c_0_8 ARRAY<INT> NOT NULL) PRIMARY KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 168 day)) DISTRIBUTED BY HASH (c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DATETIME NULL ,c_1_1 DATE NULL ,c_1_2 ARRAY<INT> NULL ,c_1_3 DOUBLE NULL ,c_1_4 DECIMAL(14, 4)  NULL ) UNIQUE KEY (c_1_0,c_1_1) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 190 day)) DISTRIBUTED BY HASH (c_1_1) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 DATE NOT NULL ,c_2_1 BOOLEAN NOT NULL ,c_2_2 DATETIME NOT NULL ,c_2_3 VARCHAR(31) NOT NULL ,c_2_4 DATETIME NOT NULL ,c_2_5 BOOLEAN NULL ,c_2_6 DATETIME NULL ,c_2_7 DATETIME NULL ,c_2_8 INT NULL ,c_2_9 FLOAT NOT NULL , INDEX index1_c_2_7(`c_2_7`) USING BITMAP) UNIQUE KEY (c_2_0,c_2_1,c_2_2,c_2_3,c_2_4,c_2_5,c_2_6,c_2_7) PARTITION BY RANGE(c_2_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 337 day)) DISTRIBUTED BY HASH (c_2_5,c_2_4,c_2_6,c_2_1,c_2_7,c_2_2,c_2_3) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t2 VALUES ("2010-01-11", false, "2010-01-31 08:00:14", "", "2010-01-12 08:00:14", false, "2009-12-31 08:00:14", "2010-01-01 08:00:14", 13, 0.1571637905791602), ("2010-01-15", true, "2010-01-17 08:00:14", "pzk", "2010-01-07 08:00:14", NULL, "2010-01-01 08:00:14", "2010-01-24 08:00:14", 4, 0.1571637905791602), ("2010-01-02", false, "2010-01-04 08:00:14", "", "2010-01-15 08:00:14", true, "2010-01-26 08:00:14", "2010-01-23 08:00:14", 24, 0.09969948155128827);
INSERT INTO t0 VALUES ("2010-01-16 08:00:14", "2010-01-07", "ZZ", 4, "k", 0.3684035740005136, "2010-01-31 08:00:14", ["9", "CdfGz", "KPB", "8", "tqm5f"], [1873625681, -870903661, 1318190170, 609858031, 1873625681, 1873625681, 695560310]);
INSERT INTO t2 VALUES ("2010-01-19", false, "2010-01-10 08:00:14", "Z", "2010-01-23 08:00:14", false, "2010-01-02 08:00:14", "2010-01-03 08:00:14", 7, 0.6925940049013147), ("2010-01-13", true, "2010-01-26 08:00:14", "L", "2010-01-29 08:00:14", false, "2010-01-31 08:00:14", "2010-01-08 08:00:14", 4, 0.8434493884936515);
INSERT INTO t1 VALUES ("2010-01-05 08:00:14", NULL, [1355683297, 1909690360, 117393213, -1160323706, -870903661, 6628637, 1532837050, 834434776, -1419311763, -945982957], 0.1952787655266014, 670973.8403), (NULL, "2010-01-21", [-333208563, 1532837050, 1600366573, -1160323706, 6628637, -2116456390, 1873625681, 6628637, -1905194718, -1546164669, 1600366573, 1909690360, 271061275], 0.04562588242324006, NULL);
INSERT INTO t0 VALUES ("2010-01-07 08:00:14", "2010-01-09", "AlWOE7g", 5, "r", 0.7817204669436614, "2010-01-14 08:00:14", ["PaOu", "", "", "Y1", "c5KK", "e8Q", "Joi6bJfb", "gbI", "8", "", "e", "jcWuBo"], [1236131491, 6628637, -945982957]), ("2010-01-01 08:00:14", "2010-01-20", "", 6, "j4acDK", 0.1571637905791602, "2010-01-10 08:00:14", ["", "U5", "Sa8"], [-764126708, -1709392663, -1202232568, 1101004821, -2116456390, 695560310, 1396104452, -2113772141, 1204523242, -2116456390]), ("2010-01-22 08:00:14", "2010-01-13", "yCks", 6, "", 0.09969948155128827, "2010-01-31 08:00:14", ["Hg", "J2OKZMd", "Z8hn", "4", "xR"], [-1227804004, 1355683297, 511464067, 800412360]);
INSERT INTO t0 VALUES ("2010-01-22 08:00:14", "2010-01-08", "KVNUQB", 6, "m", 0.973577554840304, "2010-01-06 08:00:14", ["LyeJEv", "nVCVZklA", ""], [-741744726, 1473607050, -333208563, -176267869, 2141076445, -352573213, -352573213, -1143195204, -1936357877, -1709392663, -1465395738, 868880807, -30564629, -1160323706, -2000954544]), ("2010-01-08 08:00:14", "2009-12-31", "C", 0, "n", 0.24141198884191406, "2010-01-26 08:00:14", ["cjzB1eVjr", "1rT", "LE7", "", "T", "3Z", "", "AwcD6nne", "G", "n", "XppI8AfWd", "zo5", "WZm", "HC", ""], [-609318541, 1859086068]);
INSERT INTO t1 VALUES ("2010-01-25 08:00:14", "2010-01-14", [868880807, 535529359, 117698088], 0.8199206288137129, 472977.4749);
DELETE FROM t0 WHERE c_0_0="1969-12-20 20:26:43";
INSERT INTO t1 VALUES (NULL, "2010-01-23", [690683339, -1190442497, 535529359, 695560310, 2008966444, -1805046478, -1244266200, -1874674467, -1320574138, 1019012464], 0.31029713291107386, 140313.3260);
INSERT INTO t1 VALUES ("2010-01-07 08:00:14", "2010-01-05", [-782372622, -1905194718, -180855371, 2141076445, 868569386, -2024264106, -1539062741, -1465395738, 1061317850, -1190017193, 1318190170, -1419311763], 0.524718579427207, 429574.2977), ("2010-01-23 08:00:14", "2010-01-29", [-1237034032, -2000954544], 0.6089410532358644, 140323.1850);
INSERT INTO t2 VALUES ("2010-01-01", true, "2010-01-23 08:00:14", "g7tJdUA", "2010-01-21 08:00:14", true, "2010-01-13 08:00:14", "2010-01-06 08:00:14", 31, 0.8199206288137129), ("2010-01-13", false, "2010-01-26 08:00:14", "EcP", "2010-01-05 08:00:14", true, "2010-01-02 08:00:14", "2010-01-16 08:00:14", 18, 1.355683297E9);
INSERT INTO t2 VALUES ("2010-01-19", false, "2010-01-17 08:00:14", "cSfhi", "2010-01-03 08:00:14", false, "2010-01-27 08:00:14", "2010-01-08 08:00:14", 27, 0.04562588242324006), ("2010-01-09", false, "2010-01-04 08:00:14", "", "2010-01-29 08:00:14", false, "2010-01-22 08:00:14", "2010-01-16 08:00:14", 13, 0.4906171410803889), ("2010-01-24", true, "2010-01-15 08:00:14", "", "2010-01-11 08:00:14", true, "2010-01-21 08:00:14", "2010-01-09 08:00:14", 0, -2.113772141E9);
INSERT INTO t1 VALUES ("2010-01-22 08:00:14", "2010-01-10", [1225579658, 1373004020, -372782689, 1621491008, -1875012625, -1216298466, -1190442497, 656351544, 1675585894, 1859086068, -352573213, -677733562, -147351451, 609858031, 760822383], 0.2588045649768793, 443906.4303);
CREATE VIEW v0 AS SELECT t2.c_2_0, t2.c_2_1, t2.c_2_2, t2.c_2_3, t2.c_2_4, t2.c_2_5, t2.c_2_6, t2.c_2_7, t2.c_2_8, t2.c_2_9, t0.c_0_0, t0.c_0_1, t0.c_0_2, t0.c_0_3, t0.c_0_4, t0.c_0_5, t0.c_0_6, t0.c_0_7, t0.c_0_8 FROM t2, t0 WHERE true ORDER BY t0.c_0_4, t2.c_2_9, t2.c_2_2, t0.c_0_1, t2.c_2_8, t0.c_0_0, t0.c_0_5, t2.c_2_2, t0.c_0_2, t2.c_2_3, t2.c_2_2;
INSERT INTO t1 VALUES ("2010-01-24 08:00:14", "2010-01-28", [-1831234325, -1419311763], 0.24733290953702358, 485870.6782), ("2010-01-05 08:00:14", "2010-01-26", [-677733562, -219363745, -1237034032, 1373004020, 620698934, -1601021813, 1473607050, -147351451, -945982957, 17679123], -1.874674467E9, 448534.5848);
INSERT INTO t0 VALUES ("2010-01-08 08:00:14", "2010-01-27", "", 3, "r", 0.3813813917774104, "2010-01-16 08:00:14", ["IF", "E"], [691125236, -1916341328, -915621264, -2000954544, 251668357, -1008378300, 1716050685, 395407327]), ("2010-01-06 08:00:14", "2010-01-31", "", 6, "l", 0.40436392653323516, "2010-01-05 08:00:14", ["c6", "EyrstvU", "f", "jQ6qNL1w", "", "SY", "b2", "UAol", "", "3SJLfz", "M", "X", "J"], []), ("2010-01-14 08:00:14", "2010-01-30", "PIlc", 6, "", 0.48101484328963595, "2010-01-03 08:00:14", ["", "9L", "o", "zRsg", "yG184MYkC", "", "S", ""], []);
INSERT INTO t2 VALUES ("2010-01-28", true, "2010-01-08 08:00:14", "te", "2010-01-09 08:00:14", false, "2010-01-29 08:00:14", "2010-01-18 08:00:14", 16, 0.6925940049013147), ("2010-01-16", false, "2010-01-18 08:00:14", "", "2010-01-11 08:00:14", false, "2010-01-01 08:00:14", NULL, 7, 0.5202261803115428);
INSERT INTO t0 VALUES ("2010-01-22 08:00:14", "2010-01-25", "", 6, "s", 0.6674950584119304, "2010-01-06 08:00:14", ["", "", "", "Q", "o9Lik"], []);
INSERT INTO t2 VALUES ("2010-01-12", false, "2010-01-09 08:00:14", "", "2010-01-14 08:00:14", true, "2010-01-29 08:00:14", "2010-01-03 08:00:14", 24, 7.60822383E8);
TRUNCATE TABLE t0;
-- SELECT t2.c_2_0 FROM t2, t1;
-- cardinality: 78
-- SELECT t2.c_2_0 FROM t2, t1 WHERE  ( (SELECT t0.c_0_0 FROM t0 WHERE true)  ) >= ( t2.c_2_4 )  UNION ALL SELECT t2.c_2_0 FROM t2, t1 WHERE (NOT ( ( (SELECT t0.c_0_0 FROM t0 WHERE true)  ) >= ( t2.c_2_4 ) )) UNION ALL SELECT t2.c_2_0 FROM t2, t1 WHERE (( ( (SELECT t0.c_0_0 FROM t0 WHERE true)  ) >= ( t2.c_2_4 ) ) IS NULL);
-- cardinality: 0

```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - branch-2.2 ASAN
 - mysql> select current_version();
+------------------------------+
| current_version()            |
+------------------------------+
| SQLANCER_01_ASAN 145a79c |
+------------------------------+

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5028 

## Problem Summary(Required) ：

AssertNumRow should return null if empty input.

### CODE REVIEW

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2 branch-2.1

**Review by kangkaisen — DISMISSED** [review]

File: be/src/exec/pipeline/assert_num_rows_operator.cpp [hunk 1]
```diff
@@ -11,12 +11,14 @@ namespace starrocks::pipeline {
...
     Operator::prepare(state);
 
-    // assert num rows node only use for un-correlate scalar subquery, return empty chunk is error, least fill one rows
+    // AssertNumRows should return exactly one row, report error if more than one row, return null if empty input
     vectorized::ChunkPtr chunk = std::make_shared<vectorized::Chunk>();
 
     for (const auto& desc : _factory->row_desc()->tuple_descriptors()) {
         for (const auto& slot : desc->slots()) {
-            chunk->append_column(ColumnHelper::create_column(slot->type(), true, false, 1), slot->id());
+            vectorized::ColumnPtr column = ColumnHelper::create_column(slot->type(), true, false, 1);
```
Comment (trueeyu) [inline be/src/exec/pipeline/assert_num_rows_operator.cpp:19]: does the non-pipeline engine hava the same problem?

Comment (trueeyu) [reply be/src/exec/pipeline/assert_num_rows_operator.cpp:19]: It should be that I made a mistake last time, non-ConstColumn, this size parameter is meaningless

**Review by Seaven — DISMISSED** [review]

File: be/src/exec/pipeline/assert_num_rows_operator.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exec/pipeline/assert_num_rows_operator.cpp]: fixed

**Review by trueeyu — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Comment by mofeiatwork** [comment]
@Mergify rebase

**Review by trueeyu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**